### PR TITLE
[RPS-18] Create Hippo stack

### PR DESCRIPTION
**Pull request by wojtek-oledzki**
New stacks added to startup Mysql RDS, Autoscaling Group, Load Balancer
etc.

To start up "hippo_authoring" all you need to run is:

```
make stack ROLE=hippo_authoring
```

### CODE REVIEW

**Comment by wojtek-oledzki** [comment]
ready for review @ivorc

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+  InstanceCount:
+    Description: Number of EC2 instances to launch
+    Type: Number
+    Default:  1
+
+  InstanceMonitoring:
+    Description: Should the detailed instance monitoring be enabled
+    Type: String
+    Default: false
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml:28]: Why disabled by default?

File: ansible/roles/service/hippo_authoring/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+
+hippo_authoring:
+  state: present
+  public_fqdn: "{{ public_fqdn }}"
+
+  aws:
+    instance_count: 1
+    instance_monitoring: "false"
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/defaults/main.yml:9]: Why disabled?

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+  # SSLCertificateId:
+  #   Description: SSL Certificate ARN
+  #   Type: String
+
+Resources:
+  SG:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Enable acces from VPC
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: typo: "acces" should have two S's

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+          cleanup
+
+  ASGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    Properties:
+      DesiredCapacity: !Ref InstanceCount
+      LaunchConfigurationName: !Ref LaunchConfig
+      LoadBalancerNames:
+        - !Ref LoadBalancer
+      MaxSize: 6
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: Should this be parameterised?

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+        PauseTime: PT10M
+        WaitOnResourceSignals: true
+
+  ##
+  # Loadbalancing
+  #
+  LoadBalancerSg:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Enable acces from VPC
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: Should be "access"

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+      VpcId:
+        Fn::ImportValue: !Sub "${Environment}-vpc::VpcId"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 156.67.242.53/32
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml:157]: Shall we have this as a parameter?

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+        - !Ref LoadBalancerSg
+      Subnets:
+        Fn::Split:
+          - ","
+          - Fn::ImportValue: !Sub "${Environment}-vpc::SubnetsPublic"
+      Tags:
+        - Key: Name
+          Value: !Sub "${Environment}-hippo-authoring"
+
+  ELBDNSRecordPublic:
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: Maybe it's just me, but I find this name pretty hard to parse! Would you object to using C#-style aggressive Camel Casing: ElbDnsRecordPublic? Or, even easier to read: ElbDnsPublicRecord?

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+  ScaleDownPolicy:
+    Type: AWS::AutoScaling::ScalingPolicy
+    Properties:
+      AdjustmentType: ChangeInCapacity
+      AutoScalingGroupName: !Ref ASGroup
+      Cooldown: 600
+      ScalingAdjustment: -1
+
+  CPUAlarmHigh:
```
Comment (ivorc) [inline ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: Do we actually *want* to autoscale? Remember, we'll have a fixed and limited number of licences. I'd be tempted not to autoscale based on server load.

File: ansible/roles/service/rds_mysql/defaults/main.yml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+---
+
+rds_mysql:
+  this_is_default: yes
```
Comment (ivorc) [inline ansible/roles/service/rds_mysql/defaults/main.yml]: What's this?

File: ansible/roles/service/rds_mysql/files/rds_mysql.yml [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+    Properties:
+      GroupDescription: !Sub "MySQL for ${Role}"
+      VpcId:
+        Fn::ImportValue: !Sub "${Environment}-vpc::VpcId"
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: !Ref Port
+          ToPort: !Ref Port
+          CidrIp:
+            Fn::ImportValue: !Sub "${Environment}-vpc::CidrBlock"
```
Comment (ivorc) [inline ansible/roles/service/rds_mysql/files/rds_mysql.yml:68]: Do we want to instead restrict this to the authoring and site instances, or ideally a security group only they use?

**Review by ivorc — COMMENTED** [review]
Looks good, just a few queries.

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml (hunk 1 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/hippo_authoring/files/hippo_authoring.yml:28]: $$ - we don't need it by default on all environments

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml (hunk 4 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: it should be...

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml (hunk 6 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/hippo_authoring/files/hippo_authoring.yml:157]: This is tmp solution anyway. The next version will allow us to have more dynamic list of source IPs

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml (hunk 7 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: It should have been `ElbDnsPublicRecord` :) just like everything else in the CF tpl

File: ansible/roles/service/hippo_authoring/files/hippo_authoring.yml (hunk 8 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/hippo_authoring/files/hippo_authoring.yml]: $$ savings 2 servers for more then 12h a day VS 6 servers 24/7 makes a difference in long run.

File: ansible/roles/service/hippo_authoring/defaults/main.yml (hunk 2 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/hippo_authoring/defaults/main.yml:9]: by default we don't want to generate any extra costs for non-production environments

File: ansible/roles/service/rds_mysql/defaults/main.yml (hunk 9 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/rds_mysql/defaults/main.yml]: OMG :D - testing ...

File: ansible/roles/service/rds_mysql/files/rds_mysql.yml (hunk 10 above)
Comment (wojtek-oledzki) [reply ansible/roles/service/rds_mysql/files/rds_mysql.yml:68]: yes we could make it more generic and allow to specify SgId

Comment (wojtek-oledzki) [reply ansible/roles/service/rds_mysql/files/rds_mysql.yml]: yes we could make it more generic and allow to specify SgId

**Comment by wojtek-oledzki** [comment]
@ivorc ready for another review

**Comment by ivorc** [comment]
Happy to merge as is.

**Comment by wojtek-oledzki** [comment]
ace :) Thanks
